### PR TITLE
Refresh logging

### LINKED ISSUE #212: ASPIRE Logging
ASPIRE should probably have a single dedicated logging module which wraps and configures the python logger in a consistent way.  This would be imported throughout the code in place of the python logger, and should avoid a lot of redundancy.  Probably I have some similar codes in my toolbox I can re-use to help us with this.

Some thoughts:

- Global configuration of print style/template (do we include module/file, times, etc)
- Perhaps extend the Debug, Info, Warning, Error with some extra tiers.
- Allow users to save these diagnostic logs to files (the python logger has some standard ways to tee the stream).
- Basically configurable by users though `config.ini`. (logging level, optional file names etc.)
- It might make sense to have a function in the logger module that will save the `config.ini` and some system/software information (release, if a repo, is it dirty(modified) and what is the hash then). This is both for scientific curation and potential debugging.
- Whatever paradigms we pick here need to be rationalized between things like the tutorial scripts, notebooks, and an interface (cli or ?).  We can probably design something that "should just work" for all of these things now.

### PR DESCRIPTION
**Pull request by garrettwrong**
Closes #212 .

Most of the competing logging implementation was removed in #283 , so this PR isn't too big, nice.

- Converts logger assignments to `__file__`. We may have one specific place it would be reasonable to deviate from that convention that I am aware of (the tutorials), but it is not necessary the way I have implemented logging here (timestamped logfiles).
- Convert to config file style from a a dictionary style. This moves configuration out of the source code.
- Removes duplicated logging config in `pytest.ini`, also fixes the stdout out duplication under `pytest -s`.
- Adds `fileHandler` with configurable filename.
- Tee stream, >=DEBUG to file and >=INFO to console.
- ISO datetime format in log, and close to it for filename (`:` presents a problem, replaced with `-`)
- add `levelname`

One point of discussion is regarding the default log filename.  This initial PR will drop a timestamped file name for each process where aspire package is imported.  This is pretty common, but can be a little noisy. That is, if you are frantically developing and running tests, you might get many small files in your directory to `rm *.log`, instead of one big one...  

Alternatively it is possible to turn off writing the file by default,  and use a simple bool option like `log_to_file` in `config.ini`.  This would shield users from the python logging configuration, which might seem complicated. (It isn't but certainly has nothing to do with CryoEM). This is easy to implement if desired.

As it stands the log filenames look like `aspire-2020-10-08T14-50-13.456577.log`, with a snippet example attached showing the new format. 
[ex.log](https://github.com/ComputationalCryoEM/ASPIRE-Python/files/5350689/ex.log)

### CODE REVIEW

**Comment by garrettwrong** [comment]
I went through error, warning, and info messages. Made just a few adjustments (mainly errors that should also raise). I also updated the developer guide a little based on this PR.

The only point keeping this PR a draft is regarding the log file default.  Mainly do we want multiple (individually timestamped) log files, a single log file, or no file as the default of ASPIRE.  @janden do you have any preference?  Currently the PR has the multiple timestamped setup. Thanks

**Comment by janden** [comment]
I would personally prefer to not log to file by default, but this will be very useful for bug reports. Timestamped log files I think are good for that reason too. Would a reasonable compromise be for log files to end up in a subdirectory (like `logs/`)?

**Comment by garrettwrong** [comment]
Sure I'll set that up, thanks.

File: src/aspire/utils/misc.py [hunk 1]
```diff
@@ -73,11 +85,14 @@ def src_wiener_coords(sim, mean_vol, eig_vols, lambdas=None, noise_var=0, batch_
     """
 
     if not isinstance(mean_vol, Volume):
-        logger.warning('src_wiener_coords mean_vol should be a Volume instance. Correcting for now.')
+        logger.warning('src_wiener_coords mean_vol should be a Volume instance. Attempt correction.')
```
Comment (janden) [inline src/aspire/utils/misc.py]: `logger.debug`? (Judging from similar messages above.)

**Review by janden — CHANGES_REQUESTED** [review]

File: src/aspire/utils/misc.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/utils/misc.py]: Sure, I'm hoping we get away from hitting these, but for now maybe best to leverage `debug`. We are only ones need to know atm.

**Review by janden — APPROVED** [review]

**Pull request merged**